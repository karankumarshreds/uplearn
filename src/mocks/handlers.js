import { graphql } from 'msw'

export const handlers = [
    graphql.query('GetAllJobs', (req, res, ctx) => {
        return res(
            ctx.data({
                jobs: [{
                    "id": "somerandomjobid",
                    "title": "Some random job title",
                    "cities": [
                        {
                            "id": "somerandomcityid",
                            "name": "Some city"
                        },
                        {
                            "id": "somerandomcityid2",
                            "name": "Some city 2"
                        }
                    ],
                    "countries": [
                        {
                            "id": "somerandomcityid2",
                            "name": "Some city 2"
                        }
                    ],
                    "remotes": [{
                        "name": "remote",
                        "type": "remote"
                    }],
                    "company": {
                        "name": "some comapny name",
                        "logoUrl": "//via.placeholder.com/80x80"
                    }
                }]
            })
        )
    }),
]