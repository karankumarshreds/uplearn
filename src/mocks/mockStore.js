import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { homeReducer } from "../ducks/home";
import { toastReducer } from '../ducks/toast';

const composeEnhancers = compose;

const rootReducer = combineReducers({
    home: homeReducer,
    toast: toastReducer,
});

const enhancer = composeEnhancers(applyMiddleware(thunk));
export function createTestStore() {
    return createStore(rootReducer, enhancer);
}
