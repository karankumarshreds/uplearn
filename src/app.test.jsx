import { render, screen, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";
import App from "./app";
import { createTestStore } from "./mocks/mockStore";

test("Should match snapshot", async () => {
  let renderResult;
  await act(async () => {
    renderResult = render(
      <Provider store={createTestStore()}>
        <App />
      </Provider>
    );
    return renderResult;
  });
  await waitFor(() => screen.getByTestId("jobslist"));
  expect(renderResult.container).toMatchSnapshot();
});
