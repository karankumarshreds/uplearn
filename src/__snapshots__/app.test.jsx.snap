// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <nav
    class="section"
  >
    <img
      alt=""
      class="logo"
      src="logo.jpg"
    />
  </nav>
  <main>
    <section
      data-testid="jobslist"
    >
      <article
        class="jobCard"
        data-testid="somerandomjobid"
      >
        <header
          class="jobHeader"
        >
          <img
            alt=""
            data-testid="companyLogo"
            height="80"
            src="//via.placeholder.com/80x80"
            width="80"
          />
          <div>
            <h2
              data-testid="title"
            >
              Some random job title
            </h2>
            <h3
              data-testid="company"
            >
              some comapny name
            </h3>
          </div>
        </header>
        <section
          class="meta"
        >
          <div
            class="metaInfo"
          >
            <span>
              Country:
            </span>
            <span
              data-testid="country"
            >
              Some city 2
            </span>
          </div>
          <div
            class="metaInfo"
          >
            <span>
              City:
            </span>
            <span
              data-testid="city"
            >
              Some city, Some city 2
            </span>
          </div>
          <div
            class="metaInfo"
          >
            <span>
              Offers Remote:
            </span>
            <span
              data-testid="isRemote"
            >
              Yes
            </span>
          </div>
        </section>
      </article>
    </section>
  </main>
</div>
`;
